### PR TITLE
Fix anchor diagnostics

### PR DESCRIPTION
**Pull request by kraftwerk28**
### What does this PR do?
Fixes broken warnings on unused anchors inside an array:
image

### Is it tested? How?
Test case included in PR.

### CODE REVIEW

File: test/schemaValidation.test.ts [hunk 1]
```diff
@@ -1966,5 +1966,17 @@ obj:
       expect(result[0].message).to.eq('Matches multiple schemas when only one must validate.');
       expect(telemetry.messages).to.be.empty;
     });
+
+    it('Unused anchors in an array', async () => {
```
Comment (gorkem) [inline test/schemaValidation.test.ts:1970]: This test should be in https://github.com/redhat-developer/yaml-language-server/blob/ba7452ee94105636f7c517394f378d09a71e2f6b/test/yamlValidation.test.ts#L58

**Review by gorkem — CHANGES_REQUESTED** [review]
Thank you for the contribution. If you can move the test to the correct place, it is good to go for me.

File: test/schemaValidation.test.ts (hunk 1 above)
Comment (kraftwerk28) [reply test/schemaValidation.test.ts]: Sure, I've updated an existing test case to have multiple [un]used anchors handling in an array.

**Review by gorkem — APPROVED** [review]
LGTM

**Comment by gorkem** [comment]
@msivasubramaniaan Can you review and merge this one if it is all right?

**Review by msivasubramaniaan — APPROVED** [review]
LGTM

**Pull request merged**